File: src/main/java/com/ascargon/rocketshow/api/State.java
package com.ascargon.rocketshow.api;

import com.ascargon.rocketshow.play.CompositionPlayer;
import com.ascargon.rocketshow.update.UpdateService;

import com.ascargon.rocketshow.update.UpdateState;
import jakarta.xml.bind.annotation.XmlRootElement;
import lombok.Getter;
import lombok.Setter;

@XmlRootElement
@Getter
@Setter
public class State {

    private Integer currentCompositionIndex;
    private CompositionPlayer.PlayState playState;
    private String currentCompositionName;
    private Long currentCompositionDurationMillis;
    private Long positionMillis;
    private UpdateState updateState;
    private String currentSetName;
    private String error;

}

File: src/main/webapp/src/app/models/state.ts
import { UpdateState } from "./update-state";

export class State {
    playState: string = "STOPPED";
    currentCompositionIndex: number = 0;
    currentCompositionName: string = "";
    currentCompositionDurationMillis: number = 0;
    positionMillis: number;
    updateState: UpdateState = new UpdateState();
    currentSetName: string;
    error: string;

    constructor(data?: any) {
        if (!data) {
            return;
        }

        this.playState = data.playState;
        this.currentCompositionIndex = data.currentCompositionIndex;
        this.currentCompositionName = data.currentCompositionName;
        this.currentCompositionDurationMillis = data.currentCompositionDurationMillis;
        this.positionMillis = data.positionMillis;

        if (data.updateState) {
            this.updateState = new UpdateState(data.updateState);
        }

        this.currentSetName = data.currentSetName;
        this.error = data.error;
    }
}
